src/main/java/com/google/api/codegen/transformer/ModelTypeTable.java

```diff
@@ -120,4 +121,11 @@ public String getZeroValueAndSaveNicknameFor(TypeRef type) {
   public List<String> getImports() {
     return typeTable.getImports();
   }
+
+  /**
+   * Returns the imports accumulated so far as a map of full names to short names.
+   */
+  public Map<String, String> getImportsMap() {
+    return typeTable.getImportsMap();
+  }
```

Comment:
Since some languages (here, Ruby) have more structure in their imports than the first couple languages, I think we should do a little refactoring, and create an ImportTypeView class. Changes that are necessary:

1. Delete getImportsMap(), change return type of getImports to Map<String, String>
2. For all locations that use getImports(), have them call a helper function which converts the Map<String, String> to a List<ImportTypeView>
3. Update the Java/PHP/Node snippets to get data from ImportTypeView instead of assuming that the import is just a String